.github/workflows/e2e-tests.yml

```diff
@@ -21,16 +21,13 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v4
 
-      - name: Install poetry via pipx
-        uses: abatilo/actions-poetry@v4
-        with:
-          poetry-version: 2.1.3
+      - uses: hynek/setup-cached-uv@v2
+        name: Setup uv
 
       - name: Set up Python
         uses: actions/setup-python@v5
         with:
           python-version: '3.12'
```

Comment:
This part can be left out too, to further simplify the CI as uv will install the specified python version and use it cache on succeeding runs. This is just a note and not crucial to remove and can always be done at a later point.